prepare.py

```diff
@@ -39,14 +39,25 @@ def check_ssh(ip: str) -> None:
 
     @retry(reraise=True, stop=stop_after_attempt(3), wait=wait_fixed(10))
     def connect():
-        ssh_client.connect(
-            hostname=ip,
-            username=env.USERNAME,
-            pkey=pkey,
-            look_for_keys=False,
-            allow_agent=False,
-            timeout=20,
-        )
+        if len(env.PASSWORD) > 1:
```

Comment:
you should take into account the case when `PRIVATE_KEY_PATH` is not provided and `PASSWORD` is the only secret.

Reply:
I don't understand. `PRIVATE_KEY_PATH` is always provided by env.py even if `PRIVATE_KEY` was empty.

Reply:
> PRIVATE_KEY_PATH is always provided by env.py even if PRIVATE_KEY was empty.
It doesn't make sense to provide `PRIVATE_KEY_PATH` if you have `PASSWORD` and vice versa. They are mutually exclusive arguments.


A slightly better condition here:
```python
def get_cred():
    if env.PASSWORD:
        return {"password": env.PASSWORD}
    if env.PRIVATE_KEY_PATH:
        pkey = paramiko.rsakey.RSASHA256Key.from_private_key_file(env.PRIVATE_KEY_PATH)
        return {"passphrase": env.PASSPHRASE, "pkey": pkey}
    raise ValueError("Either PASSWORD or PRIVATE_KEY_PATH must be defined")
...

ssh_client.connect(
    hostname=ip,
    username=env.USERNAME,
    look_for_keys=False,
    allow_agent=False,
    timeout=60,
    **get_cred()
)
```

Reply:
yes, you can of course write it like that. How about you merge it and change this style detail then?